src/emqx_ws_connection.erl

```diff
@@ -186,13 +187,35 @@ init(Req, Opts) ->
     case cowboy_req:parse_header(<<"sec-websocket-protocol">>, Req) of
         undefined ->
             %% TODO: why not reply 500???
-            {cowboy_websocket, Req, [Req, Opts], WsOpts};
-        [<<"mqtt", Vsn/binary>>] ->
-            Resp = cowboy_req:set_resp_header(
-                     <<"sec-websocket-protocol">>, <<"mqtt", Vsn/binary>>, Req),
-            {cowboy_websocket, Resp, [Req, Opts], WsOpts};
-        _ ->
-            {ok, cowboy_req:reply(400, Req), WsOpts}
+            case FailIfNoSubprotocol of
+                true ->
+                    {ok, cowboy_req:reply(400, Req), WsOpts};
+                false ->
+                    {cowboy_websocket, Req, [Req, Opts], WsOpts}
+            end;
+        Subprotocols ->
+            SupportedSubprotocols = proplists:get_value(supported_subprotocols, Opts),
```

Comment:
```suggestion
            SupportedSubprotocols = proplists:get_value(supported_subprotocols, Opts, []),
```

Reply:
If it is not available here, it means the code is wrong, and increasing the default value may cause the error to be hidden

Reply:
Using defaults does not affect the functionality you are trying to express, it is more likely to improve the robustness of existing code.

Reply:
This option in the current code should always have a corresponding value. In order to let us discover the missing value in time, we should not set a default value for it.